### PR TITLE
Update all the things

### PR DESCRIPTION
**Pull request by dapp**
Update kotlin version, build tools, and android plugin versions.  This will support the change to generated Kotlin code.

### CODE REVIEW

**Comment by dapp** [comment]
@fkose696  -- this should help us get the kotlin code generation stuff working.  I'd like to merge this, then I'll rebase the other stuff on top

**Review by fkose696 — DISMISSED** [review]

File: architecture/build.gradle [hunk 1]
```diff
@@ -61,7 +61,7 @@ artifacts {
 }
 
 signing {
-    sign configurations.archives
+//    sign configurations.archives
```
Comment (fkose696) [inline architecture/build.gradle:64]: do we want to keep this commented?

Comment (dapp) [reply architecture/build.gradle]: No, good catch.  Fixed

**Review by fkose696 — APPROVED** [review]
Looks good. I think we can also set exported to false like the default behavior. But since it's the sample app, it shouldn't matter

**Comment by dapp** [comment]
> 

Actually, you can't, for any activities with the LAUNCHER intent filter.   "A launchable activity must be exported as of Android 12, which also makes it available to other apps."

**Pull request merged**